src/parse.rs

```diff
@@ -167,53 +167,17 @@ pub fn parse_declaration(tokens: TokenStream) -> Result<Declaration, Error> {
                 "default" | "const" | "async" | "unsafe" | "extern" | "fn"
             ) =>
         {
-            let qualifiers = consume_fn_qualifiers(&mut tokens);
-
-            // fn keyword
-            tokens.next().unwrap();
-
-            let fn_name = consume_declaration_name(&mut tokens);
-            let generic_params = consume_generic_params(&mut tokens);
-
-            let params = match tokens.next().unwrap() {
-                TokenTree::Group(group) if group.delimiter() == Delimiter::Parenthesis => {
-                    parse_fn_params(group.stream())
-                }
-                _ => panic!("cannot parse function"),
-            };
-
-            let return_ty = consume_fn_return(&mut tokens);
-
-            let where_clause = consume_where_clause(&mut tokens);
-
-            let function_body = match &tokens.next().unwrap() {
-                TokenTree::Group(group) if group.delimiter() == Delimiter::Brace => {
-                    Some(group.clone())
-                }
-                TokenTree::Punct(punct) if punct.as_char() == ';' => None,
-                _ => panic!("cannot parse function"),
-            };
-
-            Declaration::Function(Function {
-                attributes,
-                vis_marker,
-                qualifiers,
-                name: fn_name,
-                generic_params,
-                params,
-                where_clause,
-                return_ty,
-                body: function_body,
-            })
+            let function = consume_fn(&mut tokens, attributes, vis_marker);
+            Declaration::Function(function)
         }
         Some(token) => {
             panic!(
-                "cannot parse declaration: expected keyword struct/enum/union/fn, found token {:?}",
+                "cannot parse declaration: expected keyword struct/enum/union/impl/fn, found token {:?}",
```

Comment:
Foreshadowing for #28 👻

No, this was a remnant from splitting the two PRs, I fixed it. Also ordered the keywords alphabetically (except `fn` at the end, since that's the order in which they appear in the source code).